walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,111 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // If duration is longer than a day, no option for this case.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    // Get general available time ranges.
+    Collection<String> attendees = request.getAttendees();
+    List<TimeRange> unavailableTime =
```

Comment:
Is there a way to simplify this line of code? Looks a bit confusing

Reply:
Sure. Instead of using nested helper functions, I now split the line into two separate lines.